src/Serializer.php

```diff
@@ -61,9 +79,20 @@ public function serialize(\Closure $closure)
      */
     public function unserialize($serialized)
     {
-        /** @var SerializableClosure $unserialized */
         $unserialized = unserialize($serialized);
```

Comment:
The signature should be appended or prepended to the result. It shouldn't be "re-serailized" into a new one. That's because unserializing user input can be disruptive. See http://blog.codeclimate.com/blog/2013/01/10/rails-remote-code-execution-vulnerability-explained/

Instead, consider prepending the signature, either base64 encoded or hex encoded. That way you can just substr it off .

So unserializing user input is a no-no. Instead, something like this should be done:

    if ($this->signingKey) {
        if ($signature[0] !== '%') {
            throw new \ClosureUnserializationException('Attempting to verify unsigned serialization with a key');
        }
        $signature = substr($serialized, 1, 88);
        $data = substr($serialized, 89);
        if (!hash_equals(base64_decode($signature) , hash_hmac('sha256', $data, $this->signingKey))) {
            throw new ...
        }
        $unserialized = unserialize($data);
    } elseif ($signature[0] == '%') {
        throw new \ClosureUnserializationException('Attempting to unserialize signed serialization without a key');
    }


The `%` check is just a quick way to force verification, and since it's invalid as a first character in a serialization, it shouldn't be a problem...